pandas/io/parsers/python_parser.py

```diff
@@ -431,7 +431,7 @@ def _infer_columns(self):
                         ic = len(self.index_col) if self.index_col is not None else 0
                         unnamed_count = len(this_unnamed_cols)
 
-                        if lc != unnamed_count and lc - ic > unnamed_count:
+                        if (lc != unnamed_count and lc - ic > unnamed_count) or ic == 0:
```

Comment:
Can this also impact read_fwf, read_table, and maybe others?

Reply:
I suspect that's true. I'll check for other formats and add tests for MultiIndex headers with no index.

Reply:
I've checked the other formats, but didn't add new tests because there are existing tests in place to ensure that e.g. read_table behaves the same as read_csv without duplicating every common test.

Reply:
Thanks for checking, but I don't quite follow. What tests are you referring to?

Reply:
When I wrote 

> I suspect that's true. I'll check for other formats and add tests for MultiIndex headers with no index.

I thought I would need to add tests for read_table, and read_fwf, but now I see that pandas doesn't duplicate those redundant tests. Instead, tests/io contains read_csv 653 times, read_table 31 times, and read_fwf 49 times. The tests for read_table and read_fwf are just the minimum required to show that they correctly match read_csv, and read_csv is tested in more detail. So I don't think I should add any more tests for this pull request, but that's just my best guess based on what I see so far in the code. Let me know if there's anything else to improve.

Reply:
Thanks! I didn't realize this either - makes sense.